modules/core/shared/src/main/scala/data/LTree.scala

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2018-2021 by Rob Norris
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package skunk.data
+
+import cats.Eq
+
+final case class LTree private (labels: List[String]) {
+
+  def isAncestorOf(other: LTree): Boolean = {
+    labels.isEmpty ||                 // Empty LTree is parent to all
+    other.labels.startsWith(labels)   // Other labels starts with this labels
+  }
+
+  def isDescendantOf(other: LTree): Boolean = other.isAncestorOf(this)
+  
+  override def toString: String = labels.mkString(LTree.Separator.toString())
+}
+
+object LTree {
+  val Empty = LTree(Nil)
+
+  def unsafe(labels: String*): LTree = LTree(labels.toList)
+
+  def fromString(s: String): Either[String, LTree] = {
```

Comment:
It seems like we could do all of this with a single regex.

Reply:
You're probably right. My regex-foo is limited but I can try to tackle it when I get a little more free time.